extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -3,184 +3,92 @@ require([
-		liftigniterHelpers[recircModule.id] = configuredHelper;
-
-		if (recircModule.viewModule) {
-			views[recircModule.viewModule] = views[recircModule.viewModule] || [];
-			views[recircModule.viewModule].push(deferred);
-		}
-	});
-
-	$.each(views, function (viewModule, promises) {
-		var deferred = $.Deferred();
-
-		completed.push(deferred);
-
-		log('Initializing View: ' + viewModule, 'info', logGroup);
-		require([viewModule], function (viewFactory) {
-			$.when.apply($, promises)
-				.done(function () {
-					var view = viewFactory(),
-						args = Array.prototype.slice.call(arguments),
-						data = {
-							title: '',
-							items: []
-						};
-
-					log(args, 'info', logGroup);
-
-					args.forEach(function (result) {
-						if (!result) {
-							return;
-						}
-
-						if (result.title && data.title.length === 0) {
-							data.title = result.title;
-						}
-
-						data.items = data.items.concat(result.items);
-					});
-
-					view.render(data)
-						.then(view.setupTracking())
-						.then(deferred.resolve);
-				})
-				.fail(handleError(viewModule), function () {
-					deferred.reject(viewModule);
-				});
+	// prepare & render right rail recirculation module
+	liftigniter.prepare(railRecirculation).done(function (data) {
+		require(['ext.wikia.recirculation.views.premiumRail'], function (viewFactory) {
+			viewFactory().render(data);
 		});
 	});
 
-	$.when.apply($, completed)
-		.done(function () {
-			log('Finished rendering recirculation', 'info', logGroup);
-
-			$.each(liftigniterHelpers, function (viewModule, helper) {
-				helper.setupTracking();
+	// prepare & render mixed content footer module
+	var mixedContentFooterData = [
+		liftigniter.prepare(mixedContentFooter.nsItems),
+		liftigniter.prepare(mixedContentFooter.wikiItems),
+		discussions.prepare()
+	];
+	$.when.apply($, mixedContentFooterData).done(function (nsItems, wikiItems, discussions) {
+		require(['ext.wikia.recirculation.views.mixedFooter'], function (viewFactory) {
+			viewFactory().render({
+				nsItems: nsItems,
+				wikiItems: wikiItems,
+				discussions: discussions
 			});
-		})
-		.fail(function (viewModule) {
-			log('Error running recirc at: ' + viewModule, 'info', logGroup);
 		});
+	});
 
-	function handleError(viewModule) {
-		return function (errorMessage) {
-			log(errorMessage, 'info', logGroup);
+	// fetch data for all recirculation modules
+	// TODO lazy load some data on scroll
+	liftigniter.fetch('ns');
+	liftigniter.fetch('wiki');
+	discussions.fetch();
 
-			if (viewModule === 'ext.wikia.recirculation.views.premiumRail') {
-				require([
-					'ext.wikia.recirculation.helpers.fandom',
-					viewModule
-				], function (helper, view) {
-					helper({
-						type: 'community',
-						fill: true,
-						limit: 10
-					}).loadData()
-						.done(function (data) {
-							view().render(data);
-						});
-				});
-			}
-		};
-	}
+	// TODO handle errors
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)